File: AoC_2020/scripts/script06.R
library(tidyverse)
d6 <- readLines("data/data06") %>% 
  split(cumsum(. == "")) %>% map(~.x[.x != ""])

# star 1
d6 %>% 
  map_dbl(~paste(.x, collapse = "") %>% str_split("") %>% unlist() %>%
            unique %>% length) %>% sum()

# star 2
psg <- map_dbl(d6, length)
ltr_count <- d6 %>% 
  map(~paste(.x, collapse = "") %>% str_split("") %>% unlist() %>% table())
map2_dbl(psg, ltr_count, ~sum(.x == .y)) %>% sum()

File: AoC_2022/day06/script06.py
with open("data06.txt") as d:
    data = d.read()

for i in range(4,len(data)):
    if len(set(data[i-4:i])) == 4:
        print("Star 1 =", i)
        break

for i in range(14,len(data)):
    if len(set(data[i-14:i])) == 14:
        print("Star 2 =", i)
        break